src/Aspire.Hosting.AWS/CloudFormation/CloudFormationReferenceAnnotation.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.AWS.CloudFormation;
+
+/// <summary>
+/// Resource to attach the manifest publishing callback on. The callback records in the manifest the reference between the project and the CloudFormation resource.
+/// </summary>
+/// <param name="cloudFormationResource"></param>
+/// <param name="targetResource"></param>
+internal sealed class CloudFormationReferenceResource(ICloudFormationResource cloudFormationResource, IResource targetResource) : Resource($"{targetResource.Name}-{cloudFormationResource.Name}-ref"), IResourceWithEnvironment
```

Comment:
Why do we need this?

Reply:
At the time I believe I didn't see how to augment the the manifest of a project resource besides environment variables. I thought I saw another component doing something similar to this having a reference resource. I'll take another look and see about collapse this out.

Reply:
This should just be stored in an annotation for the runtime model if you want to represent that association. You're right that we can only store that association today via environment variables but where are you planning to use this information? 

We have https://github.com/dotnet/aspire/pull/1138 but we may not have this for v1.